bokeh/document/document.py

```diff
@@ -971,6 +971,10 @@ def validate(self):
             refs = r.references()
             check_integrity(refs)
```

Comment:
The new return value return value from `check_integrity` (discussed above) will need to be stored, and then processed below according to the validation level policy.

Reply:
Given the need for the processing to *also* take in to account any `__silencers__`, the best approach is probably to encapsulate the logic for emitting warnins into a new separate function in `bokeh.validatation`. The line above should more or less then end up like:
```
validations = check_integrity(refs)

process_validations(validations)
```
which will also help prevent cluttering up `document` more.